### ISSUE TITLE
[BUG] Maker shape does not appear in legend

### ISSUE THREAD

**Issue by songhanzhang** [JuliaPlots/Plots.jl#3059]
When I draw lines along with markers, the markers are missing in the legend (see Figure 1). I think this might be a bug in the updated version, as the issue did not exist before (see Figure 2).

![bug](https://user-images.githubusercontent.com/67284799/95854843-1b52ef00-0d8a-11eb-9378-e6885feab8dc.png)
Figure 1  Drawn by the updated version

![no_bug](https://user-images.githubusercontent.com/67284799/95854854-2017a300-0d8a-11eb-99e6-1cc9316cf57d.png)
Figure 2  Drawn by the old version

Can anyone please tell me how this issue comes out, and is it possible to fix it? Thank you!

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |  x   |     |
pyplot       |     |     |
plotly       |     |     |
plotlyjs     |     |     |
pgfplotsx    |     |     |
inspectdr    |     |     |

### Versions

Plots.jl version: 1.6.10
Backend  version (`]st -m`): GR v0.52.0
Output of `versioninfo()`:
Julia Version 1.4.2
Commit 44fa15b150* (2020-05-23 18:35 UTC)

Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Xeon(R) E-2236 CPU @ 3.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 6

**Comment by BeastyBlacksmith** [comment]
> Figure 2 Drawn by the old version

Which version was the old version?

**Comment by songhanzhang** [comment]
> > Figure 2 Drawn by the old version
> 
> Which version was the old version?

Sorry, I cannot remember clearly. As far as I can remember, Figure 2 was drawn in August. The "Plots" package was added several months ago, and I did not update it until last weekend.

I just tried another computer (mac OS) where the Plots version is 1.6.3, and everything is correct.

**Comment by BeastyBlacksmith** [comment]
I can't replicate on 1.6.10 with
```julia
plot(1:5, marker=:o)
```
do you have a MWE?

**Comment by BeastyBlacksmith** [comment]
I can't replicate on 1.6.10 with
```julia
plot(1:5, marker=:o)
```
do you have a MWE?

**Comment by songhanzhang** [comment]
> I can't replicate on 1.6.10 with
> 
> ```julia
> plot(1:5, marker=:o)
> ```
> 
> do you have a MWE?

Yes, I have. Please see below. The variables k_A0_cvg, k_S0_cvg ... are calculated from my program (which is too much lengthy), you may define them as random instead. Thank you!

`fig_ex_ice0_cvg = plot(size = (560,250),`
`                       dpi = 900,`
`                       legend = true,`
`                       frame_style = :box,`
`                       grid = false,`
`                       tickfontsize = 15)`
`plot!(1:1:20, k_A0_cvg, label = "A0",`
`      color = :green3, w = 1, linestyle = :dash,`
`      shape = :circle, markersize = 5)`
`plot!(1:1:20, k_S0_cvg, label = "S0",`
`      color = :green3, w = 1, linestyle = :solid,`
`      shape = :circle, markersize = 5)`
`plot!(1:1:20, k_A1_cvg, label = "A1",`
`      color = :hotpink1, w = 1, linestyle = :dash,`
`      shape = :star, markersize = 6)`
`plot!(1:1:20, k_S1_cvg, label = "S1",`
`      color = :hotpink1, w = 1, linestyle = :solid,`
`      shape = :star, markersize = 6)`
`plot!(1:1:20, k_A2_cvg, label = "A2",`
`      color = :dodgerblue, w = 1, linestyle = :dash,`
`      shape = :diamond, markersize = 5)`
`plot!(1:1:20, k_S2_cvg, label = "S2",`
`      color = :dodgerblue, w = 1, linestyle = :solid,`
`      shape = :diamond, markersize = 5)`
`xlabel!("Number of linear elements", guidefontsize = 15)`
`ylabel!("||ϵ||")`
`plot!(xlims = (1.5,20.5), ylims = (-5,100))`
`plot!(xticks = (2:2:20))`
`plot!(bottom_margin = 3mm)`
`plot!(legendfontsize = 11)`

**Comment by songhanzhang** [comment]
I find out the reason!

In my variables k_A0_cvg, k_S0_cvg ..., the first value is NaN, e.g.:

julia> k_A0_cvg
50×1 Array{Float64,2}:
 NaN
   2.0308838226954906
   1.7986423152838897
   1.5360780314366798
   1.2788083916905897
   1.0515969494307917
   ⋮
 NaN
 NaN
 NaN
 NaN
 NaN

If I change the first value by:
`k_A0_cvg[1] = 1`
The marker then comes out in the legend.

Now I can avoid this issue by drawing piecewise lines, but it is not so convenient especially when having a large number of NaNs in a vector. So is there a better solution?

**Comment by johanmon** [comment]
jI'vce noticed the same problem, also had it down to a NaN in first position. Tried the nightly build but problem is there:

julia> using Plots
julia> using GR
julia> gr()
Plots.GRBackend()

julia> Plots.plot([missing,2,3,4,5], seriestype=:scatter)  # no shape in legend

julia> Plots.plot([1,missing,3,4,5], seriestype=:scatter)  # shape in legend ok


Julia Version 1.6.0-DEV.1699
Commit 8a5ac94e96* (2020-12-08 03:24 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: AMD Ryzen 7 4700U with Radeon Graphics
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.0 (ORCJIT, znver2)

  "GR"         => v"0.53.0"
  "Plots"      => v"1.9.1"

**Comment by Volker-Weissmann** [comment]
Simpler code to reproduce
```
using Plots
p = plot([1, NaN], [1, 2], label="bug", seriestype=:scatter)
savefig("good.pdf")

p = plot([NaN, 1], [2, 1], label="bug", seriestype=:scatter)
savefig("bad.pdf")
```
output:
[good.pdf](https://github.com/JuliaPlots/Plots.jl/files/7023036/good.pdf)
[bad.pdf](https://github.com/JuliaPlots/Plots.jl/files/7023037/bad.pdf)




```
Julia Version 1.6.2
Commit 1b93d53fc4* (2021-07-14 15:36 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, haswell)
```
